subsys/nrf_security/src/ssf_secdom/bounce_buffers.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+
+#include "bounce_buffers.h"
+
+/* k_heap_alloc allocated memory is aligned on a multiple of pointer sizes. The HW's DataUnit size
+ * must match this Zephyr behaviour.
+ */
+BUILD_ASSERT(CACHE_DATA_UNIT_SIZE == sizeof(uintptr_t));
+
+static K_HEAP_DEFINE(out_buffer_heap,
+		     ROUND_UP(CONFIG_PSA_SSF_CRYPTO_CLIENT_OUT_HEAP_SIZE, CACHE_DATA_UNIT_SIZE));
```

Comment:
Do we need to round up when we have ensured the alignment above? Could this result in spending more memory than the user has configured?

Reply:
it's if CONFIG_PSA_SSF_CRYPTO_CLIENT_OUT_HEAP_SIZE is not dividable by 4, you run risk of the linker putting something into the 'leftover' space. Unlikely, but for correctness should be done, it'll be max 3 bytes more than configured for a buffer that is probably at least 1K